### ISSUE TITLE
XOAuth Error (same as #524)

### ISSUE THREAD

**Issue by eriji** [jberkel/sms-backup-plus#539]
Original issue #524 from Pseudomous:
Hello, this was working great but suddenly got the XOAuth error. IMAP is enabled. While I do have 2-step verification enabled, the program worked fine with it before. Please help. Have deleted and reinstalled but same error... many thanks.

i have a oneplus one phone.

----------
I have a Samsung Galaxy S4
I'm having the same problem as pseudomous above.   I have 2-step verification, SMS Backup+ worked until Dec 14, then stopped working.   Have not been able to get this working since.

Get "XOAuth authorization error.  Please make sure you enabled IMAP in your Gmail account settings"

IMAP is and has been enabled.

Here is the Log:
03-08 20:08 Backup requested (manual)
03-08 20:08 Starting backup (MANUAL)
03-08 20:08 Backing up (916 SMS, 0 MMS, 163 call log)
03-08 20:08 Using config: BackupConfig{imap=BackupImapStore{uri=imap+ssl+://XOAUTH2:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@imap.gmail.com:993}, skip=false, currentTry=0, maxItemsPerSync=-1, groupToBackup=ContactGroup{_id=-1, type=EVERYBODY}, backupType=MANUAL, debug=true, typesToBackup=[SMS, MMS, CALLLOG, WHATSAPP]}
03-08 20:08 need to perform xoauth2 token refresh
03-08 20:08 error refreshing token: com.zegoggles.smssync.auth.TokenRefreshException: android.accounts.AuthenticatorException: User intervention required. Notification has been pushed., cause=android.accounts.AuthenticatorException: User intervention required. Notification has been pushed.
03-08 20:08 Authentication error: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)] (exception: com.fsck.k9.mail.store.XOAuth2AuthenticationFailedException: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)])
03-08 20:08 BackupState{currentSyncedItems=0, itemsToSync=0, backupType=MANUAL, exception=com.fsck.k9.mail.store.XOAuth2AuthenticationFailedException: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)], state =ERROR}
03-08 20:08 Backup finished
03-08 20:10 Backup requested (manual)
03-08 20:10 Starting backup (MANUAL)
03-08 20:10 Backing up (916 SMS, 0 MMS, 163 call log)
03-08 20:10 Using config: BackupConfig{imap=BackupImapStore{uri=imap+ssl+://XOAUTH2:XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX@imap.gmail.com:993}, skip=false, currentTry=0, maxItemsPerSync=-1, groupToBackup=ContactGroup{_id=-1, type=EVERYBODY}, backupType=MANUAL, debug=true, typesToBackup=[SMS, MMS, CALLLOG, WHATSAPP]}
03-08 20:10 need to perform xoauth2 token refresh
03-08 20:10 error refreshing token: com.zegoggles.smssync.auth.TokenRefreshException: android.accounts.AuthenticatorException: User intervention required. Notification has been pushed., cause=android.accounts.AuthenticatorException: User intervention required. Notification has been pushed.
03-08 20:10 Authentication error: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)] (exception: com.fsck.k9.mail.store.XOAuth2AuthenticationFailedException: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)])
03-08 20:10 BackupState{currentSyncedItems=0, itemsToSync=0, backupType=MANUAL, exception=com.fsck.k9.mail.store.XOAuth2AuthenticationFailedException: Command: *sensitive*; response: #2# [NO, [ALERT], Invalid credentials (Failure)], state =ERROR}
03-08 20:10 Backup finished

**Comment by eriji** [comment]
I did try a disconnect and reconnect.   The reconnect failed with a similar error.

**Comment by jberkel** [comment]
thanks for the detailed report. the log says `User intervention required. Notification has been pushed.` Looks like some manual user intervention is required to refresh the token.

**Comment by jberkel** [comment]
did you get a system notification when the error happened? it should be a password prompt or similar.

**Comment by eriji** [comment]
No, I was waiting for something, and looked for it, but there was no
indication anywhere that login was waiting.

Other programs using authorization might forward you to the browser to
"finish" logging in, some use the one-time tokens (which require password
entry), and most of them now just have a screen come up that says what
information is being shared, permissions, etc, and to accept.

--eric powell

--------------------------------------
eric powell
po box 129, orangeburg, sc 29116-0129
803-997-0707 - fax 877-885-8830
epowell@pobox.com

**Comment by jberkel** [comment]
Ok, that's weird. Android should handle this case (as it says in the log: Notification has been pushed.) I'll see if I can reproduce it.

**Comment by eriji** [comment]
Jan,

Something was bugging me about having seen something about Google and
XOAuth changing recently, and when I got some time I searched for it and
found some tweets by searching for "Google Authentication API".   They
indicated that there was a problem with an error similar to what I was
experiencing that caused odd authentication problems over a variety of apps.

After reading that, I tried the SMS Backup+ authentication again, and it
worked fine now.   It appears it was a Google problem and temporary.

I'm sorry that you were chasing a problem that turned out to be Google's
problem.  Thanks for responding to my inquiry so quickly.

Everything seems to work fine now.

Cheers,
--eric powell

**Comment by eriji** [comment]
Something was bugging me about having seen something about Google and XOAuth changing recently, and when I got some time I searched for it and found some tweets by searching for "Google Authentication API".   They indicated that there was a problem with an error similar to what I was experiencing that caused odd authentication problems over a variety of apps.

After reading that, I tried the SMS Backup+ authentication again, and it worked fine now.   It appears it was a Google problem and temporary.
